universal-application-tool-0.0.1/app/repository/ProgramRepository.java

```diff
@@ -54,30 +55,25 @@ public Program updateProgramSync(Program program) {
     return program;
   }
 
-  public Program createOrUpdateDraft(Program existingProgram) throws ProgramNotFoundException {
+  public Program createOrUpdateDraft(Program existingProgram) {
+    Version draftVersion = versionRepository.get().getDraftVersion();
     Optional<Program> existingDraft =
-        ebeanServer
-            .find(Program.class)
-            .where()
-            .eq("lifecycle_stage", LifecycleStage.DRAFT.getValue())
-            .eq("name", existingProgram.getProgramDefinition().adminName())
-            .findOneOrEmpty();
+        draftVersion.getProgramByName(existingProgram.getProgramDefinition().adminName());
     if (existingDraft.isPresent()) {
       Program updatedDraft =
           existingProgram.getProgramDefinition().toBuilder()
               .setId(existingDraft.get().id)
-              .setLifecycleStage(LifecycleStage.DRAFT)
               .build()
               .toProgram();
       this.updateProgramSync(updatedDraft);
       return updatedDraft;
     } else {
-      Program newDraft =
-          existingProgram.getProgramDefinition().toBuilder()
-              .setLifecycleStage(LifecycleStage.DRAFT)
-              .build()
-              .toProgram();
-      insertProgramSync(newDraft);
+      Program newDraft = existingProgram.getProgramDefinition().toBuilder().build().toProgram();
```

Comment:
question (blocking): why `.toBuilder().build()`?

Reply:
Added comment - to clear version information.